lib/internal/main/eval_string.js

```diff
@@ -24,7 +24,7 @@ markBootstrapComplete();
 
 const source = getOptionValue('--eval');
 const print = getOptionValue('--print');
-const loadESM = getOptionValue('--import').length > 0;
+const loadESM = getOptionValue('--import').length > 0 || getOptionValue('--experimental-loader').length > 0;
```

Comment:
With #48559 coming soon, should this merely check whether `--experimental-loader` was specified (rather than whether it was fed values)? After 48559, that flag need be only present to un-gate loaders and thus `module.register`.

Or should that PR handle this switch?

Reply:
I don't think that flag works without a value, does it?

Reply:
It will need to after 48559

Reply:
Depending on which PR lands first, the other one will need to adapt. We can't really do that "in advance", it shouldn't be very hard to adapt though